go/chat/server.go

```diff
@@ -2286,3 +2286,59 @@ func (h *Server) GetStaticConfig(ctx context.Context) (res chat1.StaticConfig, e
 		DeletableByDeleteHistory: chat1.DeletableMessageTypesByDeleteHistory(),
 	}, nil
 }
+
+func (h *Server) ResolveUnfurlPrompt(ctx context.Context, arg chat1.ResolveUnfurlPromptArg) (err error) {
+	var identBreaks []keybase1.TLFIdentifyFailure
+	defer func() {
+		// squash all errors coming out of here
+		err = nil
+	}()
+	ctx = Context(ctx, h.G(), arg.IdentifyBehavior, &identBreaks, h.identNotifier)
+	defer h.Trace(ctx, func() error { return err }, "ResolveUnfurlPrompt")()
+	uid, err := utils.AssertLoggedInUID(ctx, h.G())
+	if err != nil {
+		return err
+	}
+	atyp, err := arg.Result.ActionType()
+	if err != nil {
+		return err
+	}
+	fetchAndUnfurl := func() error {
+		conv, err := GetUnverifiedConv(ctx, h.G(), uid, arg.ConvID, true)
+		if err != nil {
+			return err
+		}
+		msgs, err := h.G().ConvSource.GetMessages(ctx, conv, uid, []chat1.MessageID{arg.MsgID}, nil)
+		if err != nil {
+			return err
+		}
+		if len(msgs) != 1 {
+			return errors.New("message not found")
+		}
+		h.G().Unfurler.UnfurlAndSend(ctx, uid, arg.ConvID, msgs[0])
```

Comment:
won't this cause problems since it's not idempotent? we'll just keep requeuing the same URLs if there are multiple present in the message? we should add this in the tests but i think the unfurler should have some notion of deduping before making a new task

Reply:
Hmm you could be right, let me think about it.

Reply:
Ok @joshblum, PTAL!